libcontainer/cgroups/ebpf/ebpf.go

```diff
@@ -127,12 +188,35 @@ func LoadAttachCgroupDeviceFilter(insts asm.Instructions, license string, dirFd
 		//       we fail-open on a security feature, which is a bit scary.
 		return nil
 	}
-	// If there was more than one old program, give a warning (since this
-	// really shouldn't happen with runc-managed cgroups) and then detach all
-	// the old programs.
-	if len(oldProgs) > 1 {
-		logrus.Warnf("found more than one filter (%d) attached to a cgroup -- removing extra filters!", len(oldProgs))
-		for _, oldProg := range oldProgs {
+	if !useReplaceProg {
+		// If there was more than one old program, give a warning (since this
+		// really shouldn't happen with runc-managed cgroups) and then detach
+		// all the old programs.
+		infoLevel := logrus.DebugLevel
+		if len(oldProgs) > 1 {
+			// Provide some more detailed debugging info.
+			logrus.Warnf("found more than one filter (%d) attached to a cgroup -- removing extra filters!", len(oldProgs))
+			infoLevel = logrus.InfoLevel
+		}
+		for idx, oldProg := range oldProgs {
+			// Can't use Debugf because then it won't show up during CI.
+			if info, err := oldProg.Info(); err == nil {
+				fields := logrus.Fields{
+					"type": info.Type.String(),
+					"tag":  info.Tag,
+					"name": info.Name,
+				}
+				if id, ok := info.ID(); ok {
+					fields["id"] = id
+				}
+				if runCount, ok := info.RunCount(); ok {
+					fields["run_count"] = runCount
+				}
+				if runtime, ok := info.Runtime(); ok {
+					fields["runtime"] = runtime.String()
+				}
+				logrus.WithFields(fields).Logf(infoLevel, "removing old filter %d from cgroup", idx)
```

Comment:
Is this printed to stdout or stderr?
If we are using stdout here, consumers might cause some problem

Reply:
It goes to the same place as `logrus.Debugf` et al -- so stderr? `Logf` is just a way to programattically set the log level (`Debugf` -> `Logf(logrus.DebugLevel, ...)`).

Reply:
I think `logrus.Infof(...)` (and `logrus.Logf(logrus.InfoLevel, ...)` go to stdout.

Maybe we can add a global `var debugEbpfDevice bool` which tests can set to `true`?

Reply:
It might be easier to have a `_` import of a package that has an `init` that sets the log level to debug -- though this does mean we need to be careful whenever we add the first `_test` file to a package -- but I can add a CI check for that.

Reply:
I've dropped this to Debug.